### PR TITLE
Feature 3 Added compact table display option

### LINKED ISSUE #3: [Feature Request] Compact table display
Currently, when running
```
habits tran ls
```
or 
```
habits transaction list
```
A padded table is displayed. Despite the fact that we currently have some pagination features (i.e., allowing the users to specify `-l, --limit` and `-s, --skip` options), it would still be nice to allow the display of a more compact table, so the users can see more at a time.

### PR DESCRIPTION
**Pull request by evencewang**
Added a "-c --compact" flag for tran list (ls) to show a more compact table where only Transaction ID, Title, and Amount are displayed.

### CODE REVIEW

**Comment by evencewang** [comment]
@lanyanxiang If you would like me to refactor the concise table function into the original displayTransactions function with an additional parameter I could do that as well. But if you think an extra function is fine then we can leave it and make further changes based upon that.

**Comment by jimmy-lan** [comment]
> If you would like me to refactor the concise table function into the original displayTransactions function with an additional parameter I could do that as well. But if you think an extra function is fine then we can leave it and make further changes based upon that.

These should be separate functions so I don't see any problems here. Thanks for contributing!

File: src/commands/transaction/list.ts [hunk 1]
```diff
@@ -27,10 +27,34 @@ interface PromptAnswers {
...
+  );
+  display.table.print(table);
+};
+
 const displayTransactions = (response: SuccessResponse) => {
   const transactions = response.data.payload as ListResponsePayload;
   if (!transactions.length) {
-    return console.log("Listed 0 transactions.");
+    return console.log(chalk.cyan("No transactions to display"));
   }
```
Comment (jimmy-lan) [inline src/commands/transaction/list.ts:58]: I don't see a reason for this change, I think the original message might be good enough. Just want to make sure that I'm not missing context, what triggered you to make this copy adjustment?

File: src/commands/transaction/list.ts [hunk 2]
```diff
@@ -120,11 +148,16 @@ export class ListCommand extends QuestionCommand<PromptAnswers> {
...
 
   async run(): Promise<void> {
+    const concise = this.opts.concise;
     await this._promptForProperty();
     const response = await this._sendRequest();
     if (response.isError) {
       return;
     }
-    displayTransactions(response);
+    if (!concise) {
```
Comment (jimmy-lan) [inline src/commands/transaction/list.ts:157]: I think we can just do `if (!this.opts.concise)`.

**Review by jimmy-lan — CHANGES_REQUESTED** [review]
Hi Evence,

Thanks for contributing, this is incredible! I've been procrastinating on this feature lately. Thank you so much for adding this feature!

Just some minor comments, will be nice if you can take a look. Please disregard the failing prettier workflow, It's due to the limitations of GitHub checkout action that code from forks cannot be fetched.

**Comment by jimmy-lan** [comment]
Sorry, looks like I need to adjust my bot further LOL, anyways, please feel free to use the newly created base branch to make the above adjustments.

File: src/commands/transaction/list.ts (hunk 1 above)
Comment (evencewang) [reply src/commands/transaction/list.ts:58]: Oh that is just from some experiences with other applications using such language. Not really a big deal just feels more comfortable to read. Could be changed back any time. :)

File: src/commands/transaction/list.ts (hunk 2 above)
Comment (evencewang) [reply src/commands/transaction/list.ts:157]: Absolutely!

File: src/commands/transaction/list.ts (hunk 1 above)
Comment (jimmy-lan) [reply src/commands/transaction/list.ts:58]: It'll be nice if we can change this back, to stay consistent with many tools that the developer uses.

**Pull request merged**